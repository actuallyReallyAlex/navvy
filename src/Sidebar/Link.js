import React from 'react'
import PropTypes from 'prop-types'
import { Box, Button, Text } from 'grommet'
import LinkIndicator from './LinkIndicator'
import LinkIcon from './LinkIcon'

const Link = ({ handler, icon, selected, setSelected, size, title }) => {
  const handleClick = () => {
    setSelected(title)
    handler && handler()
  }

  return (
    <Button hoverIndicator onClick={handleClick} plain>
      <Box
        direction="row"
        height="45px"
        margin={{ vertical: 'xsmall', left: 'xsmall' }}
        pad={{ vertical: 'small', left: 'small' }}
        style={{ position: 'relative' }}
        width="95%"
      >
        {selected === title && <LinkIndicator />}
        <Box
          align="center"
          direction="row"
          gap="small"
          justify={size === 'expanded' ? 'start' : 'center'}
          width={size === 'expanded' ? '100%' : '75%'}
        >
          <LinkIcon icon={icon} />
          {size === 'expanded' && <Text>{title}</Text>}
        </Box>
      </Box>
    </Button>
  )
}

Link.propTypes = {
  handler: PropTypes.func,
  icon: PropTypes.string,
  selected: PropTypes.string.isRequired,
  setSelected: PropTypes.func.isRequired,
  size: PropTypes.oneOf(['expanded', 'collapsed']).isRequired,
  title: PropTypes.string.isRequired
}

Link.defaultProps = {
  icon: 'FormSubtract'
}

export default Link
