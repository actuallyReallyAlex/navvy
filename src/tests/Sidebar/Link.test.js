import React from 'react'
import Link from '../../Sidebar/Link'
import { shallow } from 'enzyme'

const defaultProps = {
  handler: jest.fn(),
  selected: 'Title',
  setSelected: jest.fn(),
  title: 'Title'
}

const nonSelectedLinkProps = {
  handler: jest.fn(),
  selected: 'NA',
  setSelected: jest.fn(),
  title: 'Title'
}

describe('<Link />', () => {
  afterEach(() => {
    defaultProps.handler.mockClear()
  })

  test('Should render a selected Link with sidebar expanded.', () => {
    const component = shallow(<Link {...defaultProps} size="expanded" />)
    expect(component).toMatchSnapshot()
  })

  test('Should render a selected Link with sidebar collapsed.', () => {
    const component = shallow(<Link {...defaultProps} size="collapsed" />)
    expect(component).toMatchSnapshot()
  })

  test('Should render a non-selected Link.', () => {
    const component = shallow(
      <Link {...nonSelectedLinkProps} size="expanded" />
    )
    expect(component).toMatchSnapshot()
  })

  test('Should call the link handler if the link is clicked.', () => {
    const component = shallow(<Link {...defaultProps} size="expanded" />)
    expect(defaultProps.handler).not.toHaveBeenCalled()
    component.find('WithTheme(Button)').simulate('click')
    expect(defaultProps.handler).toHaveBeenCalled()
  })

  test('Should render a collapsable link.', () => {
    const component = shallow(<Link {...collapsableProps} size="expanded" />)
  })
})
