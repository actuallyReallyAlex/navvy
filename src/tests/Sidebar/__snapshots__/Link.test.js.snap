// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Link /> Should render a non-selected Link. 1`] = `ShallowWrapper {}`;

exports[`<Link /> Should render a selected Link with sidebar collapsed. 1`] = `ShallowWrapper {}`;

exports[`<Link /> Should render a selected Link with sidebar expanded. 1`] = `ShallowWrapper {}`;
